cpp/src/arrow/compute/exec/util.h

```diff
@@ -19,6 +19,8 @@
 
 #include <atomic>
 #include <cstdint>
+#include <thread>
```

Comment:
Hmm, I hate to add this in late but I didn't notice it earlier.  So far we have managed to keep `<thread>` out of the public API surface.  Is there anyway you can push this into the `util.cc`?  This utility (https://github.com/apache/arrow/blob/4591d76fce2846a29dac33bf01e9ba0337b118e9/cpp/src/arrow/util/io_util.h#L346) can probably prevent you from having to resort to pimpl.

Reply:
sure. that can be done :-)